pkg/gui/staging_panel.go

```diff
@@ -33,9 +33,13 @@ func (gui *Gui) refreshStagingPanel(forceSecondaryFocused bool, selectedLineIdx
 		gui.Views.Secondary.Title = gui.Tr.StagedChanges
 	}
 
+	// if we were ignoring whitespace, stop doing that so that
+	// line-by-line diffs work as expected
+	_ = gui.setIgnoreWhitespaceFlag(false)
```

Comment:
rather than set the state here, I would leave it as-is, but simply pass `false` into the calls to `WorktreeFileDiff` below. There's no harm in having whitespace resume being ignored once the user is out of the staging panel